### PR TITLE
Publish on releases

### LINKED ISSUE #33: Create releases with the bundles
Right now, each time that we install this library using the repository, the code is compiled and bundles of this repo are created. This makes the installation slower, maybe we could use github releases with the bundles.

I have seen that other projects use [semantic-release](https://github.com/semantic-release/semantic-release) on each push to master to create the releases, but that requires to use a conventional commits (Angular Commit Message Conventions).

### PR DESCRIPTION
**Pull request by morenol**
Publish releases to npm. Resolves #33 

This needs:

- [ ] Setup NPM_PUBLISH secret
- [x] Decide if want to rename this to something inside the @fluvio  npm scope like the other npm related packages.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+          registry-url: 'https://registry.npmjs.org'
+      - name: Prepare publish
+        run: npm ci
+      - name: Publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH }}
+        run: npm publish --access public
```
Comment (simlay) [inline .github/workflows/publish.yml:33]: Do you have thoughts on using `wasm-pack publish`? I've not actually tried it.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  push:
+    tags:
+      - 'v[0-9]+(\.[0-9]+)*'
+jobs:
+  publish:
+    name: Publish to npm
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '14' ]
```
Comment (simlay) [inline .github/workflows/publish.yml:14]: Let's go with node 16. It looks like [it's "current"](https://nodejs.org/en/about/releases/).

File: .github/workflows/publish.yml (hunk 1 above)
Comment (morenol) [reply .github/workflows/publish.yml:33]: I wasnt aware of that, I will take a look https://github.com/rustwasm/wasm-pack/blob/d46d1c69b788956160deed5e4e603f4f2780ffcf/docs/src/commands/pack-and-publish.md

Comment (morenol) [reply .github/workflows/publish.yml:33]: It looks like it runs wasm-pack pack and the publish the content of pkg/

Not sure if that is better than our rollup bundle

Comment (simlay) [reply .github/workflows/publish.yml:33]: Ah. Fair. Yeah, if this works better with our rollup bundle, let's do that.

**Pull request merged**